src/electionguard/api/filename.c

```diff
@@ -4,6 +4,113 @@
 #define __USE_SECURE_APIS__
 #endif
 
+bool generate_filename(char *path_in, char *prefix_in, char* default_prefix, char *filename_out)
+{
+    bool ok = true;
+
+    char prefix[FILENAME_MAX];
```

Comment:
Question. Why are we using the stack instead of heap for these buffers?

Reply:
only because the prefix and path arrays are short lived for the length of this method.  In the current implementation this method gets called each time a specific filename needs to be accessed.  For instance when we call `EncryptBallot` the filename is regenerated for every call, even though in our implementation the filename is the same for a given date.  One could theoretically consume the file-based API's by passing in a unique file name for each ballot.  Is that not appropriate?  maybe I'm missing a side effect somewhere